src/NHibernate/Cfg/Configuration.cs

```diff
@@ -2479,6 +2482,25 @@ private IEnumerable<IPersistentIdentifierGenerator> IterateGenerators(Dialect.Di
 			return generators.Values;
 		}
 
+		/// <summary>
+		/// Returns a quoted name(schema/catalog) if needed
+		/// </summary>
+		/// <param name="name">The schema name</param>
+		/// <param name="dialect">The instance of dicalect to use</param>
+		/// <returns></returns>
+		private string GetQuotedName(string name, Dialect.Dialect dialect)
```

Comment:
Is this necessary to quote everything?

Reply:
It's kind-of correct behavior. On the Hibernate side they started conversion to "quotable identifier" value object for names (but did not finish, as usual)